src/model.ts

```diff
@@ -518,6 +523,35 @@ export function build<TSchema extends BaseSchema, TDefaults extends Partial<TSch
     }
   );
 
+  /**
+   * @description
+   * Performs an optimized `find` to test for the existence of any document matching the filter criteria.
+   *
+   * @param filter {Filter<TSchema>}
+   * @param [options] {FindOptions<TSchema>}
+   *
+   * @returns {Promise<boolean>}
+   */
+  model.exists = async (filter, options) => {
+    // If there are any entries in the filter, we project out the value from
+    // only one of them.  In this way, if there is an index that spans all the
+    // parts of the filter, this can be a "covered" query.
+    // @see https://www.mongodb.com/docs/manual/core/query-optimization/#covered-query
+    //
+    // Note that we must explicitly remove `_id` from the projection; it is often not
+    // present in compound indexes, and mongo will automatically include it in the
+    // result unless you explicitly exclude it from the projection.
+    //
+    // If you don't pass any filter option, we instead project out the primary
+    // key, `_id` (which will override the earlier exclusion).
+    const key = Object.keys(filter)[0] || '_id';
+    const result = await model.findOne(filter, {
+      ...options,
```

Comment:
part of me wonders if we should have `...options` after `projection`, in the rare case where you might want a custom projection here. 🤷

Reply:
There are a lot of potential options that might be needed here - for example, a `readPreference`

Reply:
Yeah `readPreference` is definitely the one that came to mind here. I'm not sure any other options are helpful.

I was thinking of when the projection optimization:

1. Doesn't cover
2. Is a large array field

You might want to force an _id projection -- micro optimization, but still 🤷

Reply:
I can imagine that there are situations where being able to override the projection would be helpful, although I sort of like that the type definition here says that `projection` is not an allowed option, as it doesn't really make much sense in the general case.

Since it's not allowed at the type layer, I think I prefer to override it when it does get passed in (e.g. if we are called from an untyped file).

The caller gets some control by choosing the order of the fields in the filter (assuming you are on an engine where Object.keys() retains insertion order, which I think we can take as an assumption if you are using the Mongo SDK where things like index specifications also rely on ordering).

And of course a sophisticated caller with particular needs might choose to implement this directly as a `findOne` where they get full control over the projection and interpreting the result.

Overall, I'm happy to adjust the ordering here, as I think that it also would be totally reasonable to respect a `projection` argument when we do receive it. So this is just me sharing my thoughts from when I was implementing this and made the original choice.

Let me know if you would like me to update ☺️

Reply:
> (e.g. if we are called from an untyped file).

Great point that I admit missing. 

IMHO, implicit behavior is confusing. You either choose to enable passing `projection` and have your types conform, or disabled it and through if it was passed from an untyped code.